lib/ansible/modules/cloud/azure/azure_rm_servers.py

```diff
@@ -0,0 +1,279 @@
+    pass
+
+
+class AzureRMServers(AzureRMModuleBase):
+    """Configuration class for an Azure RM Servers resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group_name=dict(
+                type='str',
+                required=True
+            ),
+            server_name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str',
+                required=True
+            ),
+            tags=dict(
+                type='dict',
+                required=False
+            ),
+            identity=dict(
+                type='dict',
+                required=False
+            ),
+            administrator_login=dict(
+                type='str',
+                required=False
+            ),
+            administrator_login_password=dict(
+                type='str',
+                required=False
+            ),
+            version=dict(
+                type='str',
+                required=False
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group_name = None
+        self.server_name = None
+        self.parameters = dict()
+
+        self.results = dict(changed=False, state=dict())
+        self.mgmt_client = None
+        self.resource_group = None
+        self.state = None
+
+        super(AzureRMServers, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                             supports_check_mode=True,
+                                             supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            else:
+                self.parameters[key] = kwargs[key]
+
+        response = None
+        results = dict()
+        to_be_updated = False
+
+        self.mgmt_client = self.get_mgmt_svc_client(SqlManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        try:
+            resource_group = self.get_resource_group(self.resource_group_name)
+        except CloudError:
+            self.fail('resource group {} not found'.format(self.resource_group_name))
+
+        response = self.get_servers()
+
+        if not response:
+            self.log("Servers instance doesn't exist")
+            if self.state == 'absent':
+                self.log("Nothing to delete")
+            else:
+                to_be_updated = True
+        else:
+            self.log("Servers instance already exists")
+            if self.state == 'absent':
+                self.delete_servers()
```

Comment:
Does the server provide an update method? Or we cannot delete it for updating

Reply:
this is actually one of the things i am wondering about, i think perhaps for create_or_update we should try to just update by default without deleting. most of these methods will probably do exactly that. i will change this.

Reply:
" the most specific logic for each module" after my experiences with Azure Container Instance i figured out that this logic may be very difficult (or almost impossible to implement), so probably it should be left to Azure SDK / actual implementation behind REST API to decide what to do with the request.
Right now i see only 2 options:
- just call create_or_update with new parameters and let it handle the request
- delete before calling create_or_update in case when create_or_update always fails (like ACI)

Reply:
I don't the second option is possible. Consider the DB or VM resource, if the resource is deleted, how about the data? As for the first one, we may need to figure out the possibility

Reply:
yes, so i think we are down to the first option.
i am actually wondering what happens if somebody tries to change, let's say location of the server, can we just move the server to different location with databases and stuff, or it will fail?
i think we should just assume that some changes are possible and some are not.

ok, for now i just don't delete instance by default, just call create_or_udpate